samoa-api/src/main/java/org/apache/samoa/learners/classifiers/trees/LocalStatisticsProcessor.java

```diff
@@ -180,16 +208,16 @@ public Processor newProcessor(Processor p) {
    * 
    * @param computeStream
    */
-  void setComputationResultStream(Stream computeStream) {
+  public void setComputationResultStream(Stream computeStream) {
     this.computationResultStream = computeStream;
   }
 
   private AttributeClassObserver newNominalClassObserver() {
-    return (AttributeClassObserver) this.nominalClassObserver.copy();
+    return new NominalAttributeClassObserver(); //further investigate this change
   }
 
   private AttributeClassObserver newNumericClassObserver() {
-    return (AttributeClassObserver) this.numericClassObserver.copy();
+    return new GaussianNumericAttributeClassObserver();//further investigate this change
```

Comment:
@fobeligi Can you please comment on why this change was necessary, and the implications of it?